types/prefs/item.go

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package prefs
+
+import (
+	"fmt"
+
+	jsonv2 "github.com/go-json-experiment/json"
+	"github.com/go-json-experiment/json/jsontext"
+	"tailscale.com/types/opt"
+	"tailscale.com/types/ptr"
+	"tailscale.com/types/views"
+	"tailscale.com/util/must"
+)
+
+// Item is a single preference item that can be configured.
+// T must either be an immutable type or implement the [views.ViewCloner] interface.
+type Item[T any] struct {
+	preference[T]
+}
+
+// ItemOf returns an [Item] configured with the specified value and [Options].
+func ItemOf[T any](v T, opts ...Options) Item[T] {
+	if cloner, ok := any(v).(views.Cloner[T]); ok {
+		v = cloner.Clone()
+	}
+	return Item[T]{preferenceOf(opt.ValueOf(v), opts...)}
+}
+
+// ItemWithOpts returns an unconfigured [Item] with the specified [Options].
+func ItemWithOpts[T any](opts ...Options) Item[T] {
+	return Item[T]{preferenceOf(opt.Value[T]{}, opts...)}
+}
+
+// Clone returns a copy of i that aliases no memory with i.
+// It is a runtime error to call [Item.Clone] if T contains pointers
+// but does not implement [views.Cloner].
+func (i *Item[T]) Clone() *Item[T] {
```

Comment:
nit: does this need to be a pointer receiver?

Reply:
yes.

Reply:
could you elaborate why?

Reply:
Interestingly enough, even though I recall needing a pointer receiver for this and other `Clone` methods in the `prefs` package for some very specific reasons, I don't actually remember the reasons, and I couldn't easily find any code in the new package or in the `cmd/viewer` and `cmd/cloner` code generators that would require these types to have `Clone` methods with pointer receivers. So, most likely, the reasons are either no longer relevant, or I was mistaken or misremembered. Anyway, I've updated them to use value receivers, and I stand corrected.

We can change them back to pointer receivers if I remember the reasons or if a need arises. Thanks for bringing this up!